### DISCUSSION TITLE
How do I render a video with fabricjs on nodejs

### DISCUSSION THREAD

**Discussion by Rajdeepc** [fabricjs/fabric.js#8094] [Q&A] [answered]
I am trying to generate apng from a mp4 file on server side with nodejs. I was able to generate the same on client side. Is there any way we can do this via nodejs. I used jsdom but it doesn't have support for HTMLVideoElement so i have used ffmpeg to extract video frames and metadata with it. Now with these information how can i create an animated apng on the canvas?

fabric: 5.2.1
Platform NodejS

<img width="1093" alt="Screenshot 2022-07-28 at 2 00 13 PM" src="https://user-images.githubusercontent.com/8790510/181461053-83cdc40c-1d31-46a4-aece-0665701608d9.png">

**Comment by ShaMan123** [comment, marked as answer]
a very quick search on the web is the first place to look at.........
https://github.com/akalverboer/canvas2apng

fabric doesn't impl this feature
look at #7187
